relay-metrics/src/aggregator/buckets.rs

```diff
@@ -23,6 +23,15 @@ impl Buckets {
         self.queue.len()
     }
 
+    /// Inserts an entry back into the collection.
+    ///
+    /// Note: this should only be used to return values back to the queue
+    /// after removing them, otherwise use [`Self::merge`].
+    pub fn insert(&mut self, key: BucketKey, value: QueuedBucket) {
+        let _old = self.queue.push(key, value);
```

Comment:
> Shouldn't this also re-add the cost of the bucket, like merge does?

No because in this case we never subtract the cost.

> Is there a functional reason why calling merge instead of insert wouldn't work?

No not really, but it's a bit annoying API wise here, because it requires a `merge` function and cost tracking.